Improve navigation in actor IO lists, refs #10048

jraddaoui opened the pull request:
Add pagination:
- Move queries to an action so they can be reused in Ajax requests
- Modify and rename repositoryHoldings.js so it works in other IO lists
  Add 'Browse all' links:
- Add links using available params for 'Subject of' list
- Add query to show only descriptions related with the actor by an event type
  in arElasticSearchPluginQuery addAdvancedSearchFilters method

jraddaoui committed:
Use ES in actor context menu component

Move query method to a different action so it can be used in Ajax requests

jraddaoui committed:
Add Ajax pagination to actor IO lists

Changes needed to repository holdings action, template and javascript
to make it work with other IO lists with pagination in the sidebar

jraddaoui committed:
Allow browse all links to IO browse page

Add query to allow limiting the results to only those related to the
actor by an event type in arElasticSearchPluginQuery

MikeFE commented on apps/qubit/modules/actor/actions/contextMenuComponent.class.php:
```diff
@@ -29,74 +29,47 @@ class ActorContextMenuComponent extends sfComponent
+      $pager->init();
 
-      $this->relatedInfoObjects[$role] = $pager->getResults();
+      $this->lists[] = array(
```
Maybe add a comment here about what $this->lists is for. It isn't immediately obvious what it's for if you're just looking at the one file and not the entire PR for context.

MikeFE commented on apps/qubit/modules/actor/actions/relatedInformationObjectsAction.class.php:
```diff
@@ -0,0 +1,122 @@
+    $limit = sfConfig::get('app_hits_per_page', 10);
+    $culture = $this->context->user->getCulture();
+
+    if (isset($request->eventTypeId))
```
Maybe just `$resultSet = self::getRelatedInformationObjects($request->actorId, $request->page, $limit, isset($request->eventTypeId) ? $request->eventTypeId : null)` ? I guess not a big deal

jraddaoui replied:
I think'`$resultSet = self::getRelatedInformationObjects($request->actorId, $request->page, $limit, $request->eventTypeId);` could work in both cases.

MikeFE commented on js/sidebarPaginatedList.js:
```diff
@@ -2,8 +2,9 @@
-  var holdingsView = function(element)
+  var paginatedListView = function(element)
   {
+    console.log(element);
```
did you accidentally leave this in for debugging?

jraddaoui replied:
Right! Thanks Mike

jraddaoui committed:
Add comment

jraddaoui committed:
Cosmetic change

jraddaoui committed:
Remove debug log

jraddaoui committed:
Avoid get_class() over array warning

Sometimes we pass directly the ES doc array to the helper function,
this was causing a warning and breaking some JSON responses in dev mode

MikeFE commented on apps/qubit/modules/actor/actions/relatedInformationObjectsAction.class.php:
```diff
@@ -0,0 +1,115 @@
+    $limit = sfConfig::get('app_hits_per_page', 10);
+    $culture = $this->context->user->getCulture();
+
+    $resultSet = self::getRelatedInformationObjects($request->actorId, $request->page, $limit, $request->eventTypeId);
```
Does this generate a notice or warning for referring to undefined `$request->eventTypeId` though, if error_reporting(E_ALL) is set?

jraddaoui replied:
Not sure what do you mean with undefined, it's forwarding to a 404 page if the value of that parameter is not a digit and, if it's not a event type id, it will return no results.

MikeFE replied:
Oh right, my bad! I still need my morning tea :)

jraddaoui commented:
Merged
